backend/src/v1/complaint/complaint.service.ts

```diff
@@ -128,12 +128,15 @@ export class ComplaintService {
       .where("officer.user_id = :idir", { idir });
 
     const result = await builder.getOne();
-
     //-- pull the user's agency from the query results and return the agency code
-    const {
-      office_guid: { agency_code },
-    } = result;
-    return agency_code;
+    if (result.office_guid && result.office_guid.agency_code) {
```

Comment:
I think this can be shortened as per the sonar cloud warning:

if(result.office_guid?.agency_code){